src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamReadAsyncMemoryOverloadsTests.cs

```diff
@@ -400,234 +443,223 @@ End Class
         }
 
-        #endregion
-
-        #region C# - Diagnostic - Analyzer
-
         [Fact]
-        public Task CS_Analyzer_Diagnostic_ByteArray()
-        {
-            return AnalyzeCSAsync(@"
-using System;
-using System.IO;
-using System.Threading;
-class C
-{
-    public async void M()
-    {
-        using (FileStream fs = File.Open(""file.txt"", FileMode.Open))
+        public Task VB_Analyzer_NoDiagnostic_AutoCastedToMemory_CancellationToken()
         {
-            byte[] buffer = new byte[fs.Length];
-            await fs.ReadAsync(buffer, 0, (int)fs.Length);
-        }
-    }
-}
-            ", GetCSResult(12, 19, 12, 58));
+            return AnalyzeVBAsync(@"
+Imports System
+Imports System.IO
+Imports System.Threading
+Class C
+    Public Async Sub M()
+        Using s As FileStream = File.Open(""path.txt"", FileMode.Open)
+            Dim buffer As Byte() = New Byte(s.Length - 1) {}
+            Await s.ReadAsync(buffer, New CancellationToken())
+        End Using
+    End Sub
+End Class
+            ");
         }
 
         [Fact]
-        public Task CS_Analyzer_Diagnostic_Inline()
-        {
-            return AnalyzeCSAsync(@"
-using System;
-using System.IO;
-using System.Threading;
-class C
-{
-    public async void M()
-    {
-        using (FileStream fs = File.Open(""file.txt"", FileMode.Open))
+        public Task VB_Analyzer_NoDiagnostic_UnsupportedVersion()
         {
-            await fs.ReadAsync(new byte[fs.Length], 0, (int)fs.Length);
-        }
-    }
-}
-            ", GetCSResult(11, 19, 11, 71));
+            return AnalyzeVBUnsupportedAsync(@"
+Imports System
+Imports System.IO
+Imports System.Threading
+Class C
+    Public Async Sub M()
+        Using s As FileStream = File.Open(""file.txt"", FileMode.Open)
+            Dim buffer As Byte() = New Byte(s.Length - 1) { }
+            Await s.ReadAsync(buffer, 0, s.Length)
+        End Using
+    End Sub
+End Class
+            ");
         }
 
-        [Fact]
-        public Task CS_Analyzer_Diagnostic_CancellationToken()
-        {
-            return AnalyzeCSAsync(@"
-using System;
-using System.IO;
-using System.Threading;
-class C
-{
-    public async void M()
-    {
-        using (FileStream fs = File.Open(""file.txt"", FileMode.Open))
-        {
-            byte[] buffer = new byte[fs.Length];
-            await fs.ReadAsync(buffer, 0, (int)fs.Length, new CancellationToken());
-        }
-    }
-}
-            ", GetCSResult(12, 19, 12, 83));
-        }
+        #endregion
 
-        [Fact]
-        public Task CS_Analyzer_Diagnostic_ConfigureAwait()
-        {
-            return AnalyzeCSAsync(@"
+        #region C# - Diagnostic - Analyzer
+
+        [Theory]
+        [InlineData("buffer, 0, (int)s.Length",
+                    "buffer.AsMemory(0, (int)s.Length)",
+                    false, 56)]
+        [InlineData("buffer, 0, (int)s.Length",
+                    "buffer.AsMemory(0, (int)s.Length)",
+                    true, 56)]
```

Comment:
Does it make sense to add an additional test parameter `withAwait` and an additional `InlineData(withAwait: false)` to verify that no diagnostic is reported when `withAwait` is `false`? Or do you feel this is already adequately tested in prior tests to not warrant adding that scenario to this test? Anyway is fine to me, this is just a suggestion in case you feel it will solidify your tests..

Reply:
I think it's sufficiently tested in the tests that are not expected to detect a diagnostic.